cg/store/api/findbasicdata.py

```diff
@@ -89,9 +93,17 @@ def customers(self) -> List[Customer]:
         """Fetch all customers."""
         return self.Customer.query
 
-    def collaboration(self, internal_id: str) -> Collaboration:
+    def _get_collaboration_query(self) -> Query:
+        """Returns a collaboration query"""
+        return self.Collaboration.query
+
+    def get_collaboration_by_internal_id(self, internal_id: str) -> Collaboration:
         """Fetch a customer group by internal id from the store."""
-        return self.Collaboration.query.filter_by(internal_id=internal_id).first()
+        return apply_collaboration_version_filter(
+            collaborations=self._get_collaboration_query(),
+            functions=[CollaborationFilters.get_collaboration_by_internal_id],
```

Comment:
Maybe a better name is ```filter_functions``` instead of ```functions``` to be super explicit.

Reply:
I agree with Sebastian to be more explicit.
I have encountered that I need filter functions from two different models, for example when there is a join.

I would argue for collaboration_filter_functions since it would cover the example mentioned above.

Reply:
I prefer `filter_functions`. Regarding your example @ChrOertlin I think the naming can be more verbose in those functions.

Reply:
Ok, then in cases where one model gets filtered: filter_functions
In cases where multiple models: <model>_filter_functions